### ISSUE TITLE
ClamAV in k8s using TCP Socket stuck in "Socket for clamd not found yet" loop

### ISSUE THREAD

**Issue by pedro-joaquim-bsurance** [Cisco-Talos/clamav#757]
Describe the bug
----------------
TL;DR - Last docker image named "stable" and "stable_base" is not working correctly when serving through TCP Socket inside k8s cluster, even though it creates `clamd.socket` file in the correct place and serves requests from other apps in the specified port, it keep trying to "find" the socket and logging `Socket for clamd not found yet, retrying` until it reaches the max retries and kill the entire container

Long Version

Since the update of the last version , docker image `clamav/clamav:stable_base` digest - `sha256:c4f6747be3b72f9263e0c9ec10d3947f1d64a9fc8c6183bde2d70acd5578ad22` the service gets stucked in looking for the clamd Socket even though it is created right after

After starting the clamAV service it  shows this (which is normal since it doesn't exist at start up) 
```
Socket for clamd not found yet, retrying (0/1800) ...Wed Nov 16 17:39:59 2022 -> ClamAV update process started at Wed Nov 16 17:39:59 2022
```

Right after we can see that the socket is created 
```
Wed Nov 16 17:39:59 2022 -> Localserver: Creating socket directory: /run/clamav
```

After this, the container is serving correctly requests done to the TCP socket and responds norrmally

but the clamd never actually finds it again for some reason, so just gets stucked until it runs out of retries, at which point it kills the container (and pod on k8s restarts, doing it all over again)
```
yet, retrying (295/1800) ... Socket for clamd not found yet, retrying (296/1800) ... Socket for clamd not found yet, retrying (297/1800) ... Socket for clamd not found yet, retrying (298/1800) ... Socket for clamd not found yet, retrying (299/1800) ... Socket for clamd not found yet, retrying (300/1800) ... Socket
```
 keeps going on...

This is not desirable because these restart mean that there is a time window where the clamAV is unreachable

Also, I tested with an older version and this does not happen in older version

I have tried the version `clamav/clamav:0.105.1-1_base` and this one works normally, start the same way, but just a couple of retries later the log saying the socket was created, the clamd "finds" the socket and proceeds  to work normally
```
2022-11-16 20:08:04 | Socket for clamd not found yet, retrying (0/1800) ... Socket for clamd not found yet, retrying (1/1800) ... Socket for clamd not found yet, retrying (2/1800) ...
(...)
2022-11-16 20:08:04 | socket found, clamd started.
2022-11-16 20:08:04 | Starting Freshclamd
```

How to reproduce the problem
----------------------------

Deploy the latest base image (or this digest to be exact - `sha256:c4f6747be3b72f9263e0c9ec10d3947f1d64a9fc8c6183bde2d70acd5578ad22` to a k8s cluster configured to serve through TCP socket
Deploy this yaml manifest to a k8s cluster
```
---
apiVersion: v1
kind: Service
metadata:
  name: clamav-service
  labels: {app: clamav, component: antivirus, env: test}
spec:
  selector: {app: clamav, component: antivirus, env: test}
  ports:
  - name: api
    port: 3310
    protocol: TCP

---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: clamav
  labels: {app: clamav, component: antivirus, env: test}
spec:
  replicas: 1
  selector: {matchLabels: {app: clamav, component: antivirus, env: test}}
  serviceName: clamav-service
  template:
    metadata:
      labels: {app: clamav, component: antivirus, env: test}
    spec:
      serviceAccountName: default
      containers:
      - name: clamav-scanner
        image: clamav/clamav:stable_base
        env: 
          - {name: CLAMAV_NO_FRESHCLAMD, value: "false"} # This env can be used to deactivate automatic updates to the AV Database (useful for debugging)
          - {name: CLAMAV_NO_MILTERD, value: "true"} # Disables clamav-milter, this is an email AV
        ports:
        - name: api
          containerPort: 3310
        volumeMounts:
          - name: clamd
            mountPath: /etc/clamav/clamd.conf
            subPath: clamd.conf
            readOnly: true
          - name: freshclam
            mountPath: /etc/clamav/freshclam.conf
            subPath: freshclam.conf
            readOnly: true
          - name: clamav-db
            mountPath: /var/lib/clamav
          - name: lock
            mountPath: /run/lock
        resources:
          requests: {memory: "1Gi", cpu: "50m"}
          limits: {memory: "3Gi", cpu: "800m"}
      volumes: 
        - name: clamd
          configMap:
            name: clamd-configmap
        - name: freshclam
          configMap: 
            name: freshclam-configmap
        # EmptyDir is to allow the folder where the DB is stored to persist through restarts and crashes
        # as to avoid having to download the entire DB after such events
        - name: clamav-db
          emptyDir: {}
        - name: lock
          emptyDir: {}
```

The config file used with only the lines that are being used (uncommented)
Config for clamd:
```
    LogFile /var/log/clamd.log
    LogFileMaxSize 25M
    LogTime yes
    LogVerbose yes
    LogRotate yes
    ExtendedDetectionInfo yes
    PidFile /run/lock/clamd.pid
    DatabaseDirectory /var/lib/clamav
    LocalSocket /run/clamav/clamd.sock
    TCPSocket 3310
    Bytecode yes
    BytecodeSecurity TrustSigned
```

config for freshclam
```
    DatabaseDirectory /var/lib/clamav
    LogFileMaxSize 25M
    LogTime yes
    LogSyslog yes
    LogRotate yes
    DatabaseMirror db.de.ipv6.clamav.net 
    DatabaseMirror database.clamav.net
    NotifyClamd /etc/clamav/clamd.conf
    TestDatabases yes
    Bytecode yes
```

This is the config inside the pod given by the command clamconf -n
PS: `database information` is lacking because with all the restart it got timed out

```
/ # clamconf -n
Checking configuration files in /etc/clamav

Config file: clamd.conf
-----------------------
LogFile = "/var/log/clamd.log"
LogFileMaxSize = "26214400"
LogTime = "yes"
LogVerbose = "yes"
LogRotate = "yes"
ExtendedDetectionInfo = "yes"
PidFile = "/run/lock/clamd.pid"
LocalSocket = "/run/clamav/clamd.sock"
TCPSocket = "3310"

Config file: freshclam.conf
---------------------------
LogFileMaxSize = "26214400"
LogTime = "yes"
LogSyslog = "yes"
LogRotate = "yes"
DatabaseMirror = "db.de.ipv6.clamav.net", "database.clamav.net"

clamav-milter.conf not found

Software settings
-----------------
Version: 0.105.1
Optional features supported: MEMPOOL AUTOIT_EA06 BZIP2 LIBXML2 PCRE2 ICONV JSON RAR

Database information
--------------------
Database directory: /var/lib/clamav
bytecode.cvd: version 333, sigs: 92, built on Mon Mar  8 15:21:51 2021
Total number of signatures: 92

Platform information
--------------------
uname: Linux 5.4.209-116.367.amzn2.x86_64 #1 SMP Wed Aug 31 00:09:52 UTC 2022 x86_64
OS: Linux, ARCH: x86_64, CPU: x86_64
zlib version: 1.2.13 (1.2.13), compile flags: a9
platform id: 0x0a21979708000000000b0201

Build information
-----------------
GNU C: 11.2.1 20220219 (11.2.1)
sizeof(void*) = 8
Engine flevel: 151, dconf: 151
```

**Comment by micahsnyder** [comment]
@pedro-joaquim-bsurance I am sorry for the trouble. This issue is actively being worked on, and is tracked here https://github.com/Cisco-Talos/clamav/issues/754

We will have a fix within the day and hopefully within an hour or two.
